### ISSUE TITLE
MusicBERT: Symbolic Music Understanding with Large-Scale Pre-Training

### ISSUE THREAD

**Issue by InabaTatsuro** [InabaTatsuro/papers#28]
# MusicBERT: Symbolic Music Understanding with Large-Scale Pre-Training [[ACL finding21](https://aclanthology.org/2021.findings-acl.70/)]

## Abstract
- MusicBERT, a large pre-training model for music understanding
- use several mechanisms, OctupleMIDI encoding and bar-level masking strategy, to enhance symbolic music data
- experimented on 4 tasks, melody completion, accompaniment suggestion, genre classification, and style classification

## Background

![スクリーンショット 2023-11-03 22 10 19](https://github.com/InabaTatsuro/papers/assets/102784221/fa22365b-3433-4cf8-be36-6b79852533a1)

- CP-like and REMI-like representations are too long to process the transformer model

### Model
![スクリーンショット 2023-11-03 22 30 14](https://github.com/InabaTatsuro/papers/assets/102784221/a94b632d-8d98-47a6-8e2c-b019d2c77567)

- Transformer encoder with masked language modeling
- concatenate the embeddings of the 8 elements and use a linear layer to convert them into a single vector
- add 8 different softmax layers to map the hidden of the transformer encoder to the vocabulary sizes of 8 elements

### OctupleMIDI encoding
![スクリーンショット 2023-11-03 22 48 19](https://github.com/InabaTatsuro/papers/assets/102784221/e8e7adae-4411-4038-a45e-62682142ceb1)

- OctupleMIDI encodes each note into a tuple with 8 elements

### Masking Strategy

![スクリーンショット 2023-11-03 22 51 39](https://github.com/InabaTatsuro/papers/assets/102784221/7b8863ff-e02d-4f7f-a072-a0542a7a1d15)

- use bar-level masking to avoid information leakage shown in the (a) OctupleMIDI encoding

### Pre-training Corpus
![スクリーンショット 2023-11-03 22 59 42](https://github.com/InabaTatsuro/papers/assets/102784221/071472b6-c80f-451e-a70f-fe4483e36749)

- crawl, clean and deduplicate

## Memo
- How is positional encoding?

**Comment by InabaTatsuro** [comment]
## Task
### Melody completion
- find the most matched consecutive phrase in a given set of candidates for a given melodic phrase
- HITS@k

### Accompaniment suggestion
- find the most related accompaniment phrase in a given set of harmonic phrase candidates for a given melodic phrase
- HITS@k

### Genre and Style classification
